tests/cpld_gridgen/ftst_find_angq.F90

```
@@ -1,108 +1,335 @@
+    call find_angchk((/2,4/),(/3,4/),angle(:,:,1),angchk(:,:,1))
+    call find_angchk((/2,4/),(/3,4/),angle(:,:,2),angchk(:,:,2))
+    call find_angchk((/2,4/),(/3,4/),angle(:,:,3),angchk(:,:,3))
+    ! reverse angle for MOM6
+    angchk = -angchk
+
+    do k = 1,nblocks
+       write(ck,'(i1.1)')k
+       dmax = 1.0e-30
+       do j = 1,nj
+          do i = 1,ni
+             if (abs(angchk(i,j,k)) .gt. 0.0) then
+                diff = abs(-angchk(i,j,k) - cice6_anglet(i,j,k))
+                dmax = max(diff,dmax)
+             end if
+          end do
+       end do
+       call passfail(dmax, 1.0e-4, 'angchk vs CICE6 anglet, block '//trim(ck))
+    end do
+
+    do k = 1,nblocks
+       write(ck,'(i1.1)')k
+       dmax = 1.0e-30
+       do j = 1,nj
+          do i = 1,ni
+             if (abs(angchk(i,j,k)) .gt. 0.0) then
+                diff = abs(angchk(i,j,k) - asin(mom6_sinrot(i,j,k)))
+                dmax = max(diff,dmax)
+             end if
+          end do
+       end do
+       call passfail(dmax, 1.2e-2, 'angchk vs MOM6 anglet, block '//trim(ck))
+    end do
+
+    if (debug) then
+       print *,'angchk'
+       print *,'left'
+       do j = 1,nj
+          print '(5f15.6)',(angchk(i,j,1), i = 1,ni)
+       end do
+       print *,'right'
+       do j = 1,nj
+          print '(5f15.6)',(angchk(i,j,2), i = 1,ni)
+       end do
+       print *,'quad'
+       do j = 1,nj
+          print '(5f15.6)',(angchk(i,j,3), i = 1,ni)
+       end do
+    end if
+  end program
+
+  subroutine passfail(dmax,tolerance,msg)
+
+    use gengrid_kinds, only : dbl_kind
+
+    implicit none
+
+    real(dbl_kind),    intent(in) :: dmax
+    real(dbl_kind),    intent(in) :: tolerance
+    character(len=*),  intent(in) :: msg
+
+    if (dmax .le. tolerance) then
+       print '(a,2f12.8)','SUCCESS! '//trim(msg)//'  ',dmax,tolerance
+    else
+       print '(a,2f12.8)','FAIL! '//trim(msg)//'  ',dmax,tolerance
```

Need to uncomment the 'stop 1' or this test will silently fail.

---

Thanks!